lib/MPIPreferences/src/MPIPreferences.jl

```diff
@@ -222,32 +222,52 @@ function identify_abi(libmpi)
         if (m = match(r"CRAY MPICH version (\d+.\d+.\d+)", version_string)) !== nothing
             version = VersionNumber(m.captures[1])
         end
+
     elseif startswith(version_string, "FUJITSU MPI")
         impl = "FujitsuMPI"
         # "FUJITSU MPI Library 4.0.0 (4.0.1fj4.0.0)\0"
         if (m = match(r"^FUJITSU MPI Library (\d+.\d+.\d+)", version_string)) !== nothing
             version = VersionNumber(m.captures[1])
         end
+
     elseif startswith(version_string, "MPIwrapper")
         impl = "MPIwrapper"
         # MPIwrapper 2.2.2
         if (m = match(r"^MPIwrapper Version:\t(\d+.\d+.\d+\w*)", version_string)) !== nothing
             version = VersionNumber(m.captures[1])
         end
+
+    elseif startswith(version_string, "HPE MPT")
+        impl = "HPE MPT"
+        # HPE MPT 2.23  08/26/20 02:54:49-root
+        if (m = match(r"^HPE MPT (\d+.\d+)", version_string)) !== nothing
+            version = VersionNumber(m.captures[1])
+        end
+
+    elseif startswith(version_string, "HPE HMPT")
+        impl = "HPE HMPT"
+        # HPE HMPT 2.23  08/26/20 02:59:48-root
+        if (m = match(r"^HPE HMPT (\d+.\d+)", version_string)) !== nothing
+            version = VersionNumber(m.captures[1])
+        end
     end
+
     # 3) determine the abi from the implementation + version
     if (impl == "MPICH" && version >= v"3.1" ||
         impl == "IntelMPI" && version > v"2014" ||
         impl == "MVAPICH" && version >= v"2" ||
-        impl == "CrayMPICH" && version >= v"7")
+        impl == "CrayMPICH" && version >= v"7" ||
         # https://www.mpich.org/abi/
```

Comment:
The comment should go at the line below?

Reply:
Not sure. The website specifically lists which MPI libraries starting which version are compatible to MPICH. HMPT is *not* listed on that website, however, I know from the docs that it is supposed to be compatible. Thus I tried to exclude it by putting it below the existing comment, but I can also move it up again. What do you think?